platform/mbed_wait_api_no_rtos.c

```diff
@@ -31,8 +31,8 @@ void wait_ms(int ms) {
 
 void wait_us(int us) {
     const ticker_data_t *const ticker = get_us_ticker_data();
-    uint32_t start = ticker_read(ticker);
-    while ((ticker_read(ticker) - start) < (uint32_t)us);
+    uint64_t start = ticker_read_us(ticker);
```

Comment:
Could you use the type `us_timestamp_t` instead of `uint64_t`  ?

Reply:
Sure, I'll change it

Reply:
Updated according @pan- suggestions